python/graphstorm/inference/np_infer.py

```diff
@@ -90,8 +90,7 @@ def infer(self, loader, save_embed_path, save_predict_path=None,
         sys_tracker.check('save embeddings')
 
         if save_predict_path is not None:
-            os.makedirs(save_predict_path, exist_ok=True)
```

Comment:
It is better to have a separate PR to do that.

Reply:
> Are you going to save the prediction results in a distributed way?

The main branch codes will save prediction results in a distributed way because it does not use (if rank==0) condition to save the predictions.

Reply:
> It is better to have a separate PR to do that.

As the issue is the same, I put them into one PR. If necessary, I can have another PR.